### PR TITLE
Package request: HPCG

### PR DESCRIPTION
**Pull request by yellowhat**
Hi,
I would like to add a new package `hpcg`.

https://www.hpcg-benchmark.org

Thanks

### CODE REVIEW

**Comment by yellowhat** [comment]
I do not think I can shorter the line 34:
```
    version('3.1', sha256='33a434e716b79e59e745f77ff72639c32623e7f928eeb7977655ffcaade0f4a4')
```
Any idea?

**Comment by michaelkuhn** [comment]
The `version` lines do not have to be shorted than 80 characters. The remaining warnings I see are in lines 33 and 34: https://github.com/spack/spack/blob/80cca50a95b1ba31c4e0425d164719af091d6391/var/spack/repos/builtin/packages/hpcg/package.py#L33-L34

File: var/spack/repos/builtin/packages/hpcg/package.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    patch('missing_variable.patch')
+
+    depends_on('mpi@1.1:')
+
+    arch = '{0}-{1}'.format(platform.system(), platform.processor())
+    build_targets = ['arch={0}'.format(arch)]
+
+    def configure(self, spec, prefix):
+        CXXFLAGS = '-O3 -ffast-math -ftree-vectorize -ftree-vectorizer-verbose=0 '
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/hpcg/package.py]:
```suggestion
        CXXFLAGS = '-O3 -ffast-math '
        CXXFLAGS += '-ftree-vectorize -ftree-vectorizer-verbose=0 '
```
This line is still too long. Maybe something like this?

File: var/spack/repos/builtin/packages/hpcg/package.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    homepage = "https://www.hpcg-benchmark.org"
+    url = "http://www.hpcg-benchmark.org/downloads/hpcg-3.1.tar.gz"
+    git = "https://github.com/hpcg-benchmark/hpcg.git"
+
+    version('develop', branch='master')
+    version('3.1', sha256='33a434e716b79e59e745f77ff72639c32623e7f928eeb7977655ffcaade0f4a4')
+
+    variant('openmp', default=True, description='Enable OpenMP support')
+
+    patch('missing_variable.patch')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/hpcg/package.py]:
```suggestion
    patch('https://github.com/hpcg-benchmark/hpcg/commit/e9e0b7e6cae23e1f30dd983c2ce2d3bd34d56f75.patch', sha256='23b9de83042eb7a8207fdddcfa79ae2cc1a17e8e623e2224c7751d7c328ee482', when='%gcc@9:')
```
This way we do not have to add the full patch to the repository.

According to the comment at https://github.com/hpcg-benchmark/hpcg/commit/e9e0b7e6cae23e1f30dd983c2ce2d3bd34d56f75, this doesn't work with older GCCs, so we should probably add a `when` clause.

File: var/spack/repos/builtin/packages/hpcg/package.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+            'ARFLAGS       = r',
+            'RANLIB        = echo',
+        ]
+
+        # Write configuration options to include file
+        with open('setup/Make.{0}'.format(self.arch), 'w') as makefile:
+            for var in config:
+                makefile.write('{0}\n'.format(var))
+
+        return
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/hpcg/package.py]:
```suggestion
```
This can be removed.

**Review by michaelkuhn — COMMENTED** [review]
Just two minor comments, otherwise this LGTM.

**Review by michaelkuhn — APPROVED** [review]

**Pull request merged**